### PR TITLE
Thread proofs through arith channels & similar

### PR DESCRIPTION
**Pull request by alex-ozdemir**
Thread proofs through conflict/lemma storages & channels in TheoryArithPrivate
* d_acTmp
* d_approxCuts
* lemma output channel
* conflict output channel

### CODE REVIEW

File: src/theory/arith/theory_arith_private.h [hunk 1]
```diff
@@ -713,8 +713,8 @@ class TheoryArithPrivate {
     return (d_containing.d_valuation).getSatValue(n);
   }
 
-  context::CDQueue<Node> d_approxCuts;
-  std::vector<Node> d_acTmp;
+  context::CDQueue<TrustNode> d_approxCuts;
```
Comment (ajreynol) [inline src/theory/arith/theory_arith_private.h:716]: Can you add some quick doc on these members?

**Review by ajreynol — CHANGES_REQUESTED** [review]
One minor doc request

File: src/theory/arith/theory_arith_private.h (hunk 1 above)
Comment (alex-ozdemir) [reply src/theory/arith/theory_arith_private.h:717]: I have rather limited understanding of how these are used, but I documented the (very little) that I know in aa52080e0

Comment (ajreynol) [reply src/theory/arith/theory_arith_private.h:717]: No worries :)

**Review by ajreynol — APPROVED** [review]

**Pull request merged**